### PR TITLE
[Zwave] Multilevel switch support (Dimmer and Curtains)

### PR DESCRIPTION
**Pull request by sescandell**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
~- [ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.
~- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

This PR is to manage multilevel switch. These devices could have multiple purpose. It handles specific curtains and standard dimmers.

Dashboard
image

Intégration
image

Tested on real life devices with Fibaro Walli Roller Shutter & Fibaro Walli Dimmer

This update introduces a new way to add multiple features from the same zWave property. This is usefull for example to manage, with the same property (currentValue) two different features: a Dimmer and a Binary state. For Dimmers, turning the device ON executes a restorePrevious  state (cloning real button behavior).

### CODE REVIEW

**Comment by William-De71** [comment]
Thanks for checking the forum, I've added some comments about this PR.

https://community.gladysassistant.com/t/evolution-du-service-z-wave-js-ui/8692/38

File: server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
- */
-function getNode(nodes, nodeId) {
-  return nodes.find((n) => n.external_id === nodeId);
+  return (
+    get(
+      ACTIONS,
+      `${cleanNames(nodeFeature.command_class_name)}.${zwaveJsNode.deviceClass.generic}-${
+        zwaveJsNode.deviceClass.specific
+      }.${baseCommandPath}`,
+    ) || get(ACTIONS, `${cleanNames(nodeFeature.command_class_name)}.${baseCommandPath}`)
```
Comment (Pierre-Gilles) [inline server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js]: Maybe let's put this in a util function?

File: server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js [hunk 2]
```diff
@@ -49,57 +52,96 @@ function getNodeFeature(node, nodeFeatureId) {
...
-        endpoint: nodeFeature.endpoint,
-      },
-      command.getName(nodeFeature),
-      commandArgs,
-    ],
-  };
-  this.publish('zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/api/sendCommand/set', JSON.stringify(mqttPayload));
+  if (action.stateUpdate) {
+    const promises = [];
+    action.stateUpdate.forEach((stateUpdate) => {
```
Comment (Pierre-Gilles) [inline server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js]: Maybe let's replace this with Bluebird Promise.map with a concurrency?

File: server/services/zwavejs-ui/utils/convertToGladysDevice.js [hunk 3]
```diff
@@ -13,13 +13,42 @@ const cleanNames = (text) => {
     .toLowerCase();
 };
 
-const getDeviceFeatureId = (nodeId, commandClassName, endpoint, propertyName, propertyKeyName) => {
+const getDeviceFeatureId = (nodeId, commandClassName, endpoint, propertyName, propertyKeyName, exposedName) => {
   const propertyKeyNameClean = cleanNames(propertyKeyName);
   return `zwavejs-ui:${nodeId}:${endpoint}:${cleanNames(commandClassName)}:${cleanNames(propertyName)}${
     propertyKeyNameClean !== '' ? `:${propertyKeyNameClean}` : ''
-  }`;
+  }${exposedName !== '' ? `:${exposedName}` : ''}`;
```
Comment (Pierre-Gilles) [inline server/services/zwavejs-ui/utils/convertToGladysDevice.js]: featureName?

File: server/services/zwavejs-ui/lib/zwaveJSUI.onNodeValueUpdated.js [hunk 4]
```diff
@@ -6,42 +6,72 @@ const { cleanNames, getDeviceFeatureId } = require('../utils/convertToGladysDevi
...
-    await this.gladys.event.emit(EVENTS.DEVICE.NEW_STATE, {
-      device_feature_external_id: nodeFeature.external_id,
-      state: convertedValue,
-    });
+  if (!valueConverters) {
+    return Promise.resolve();
   }
+
+  const promises = [];
+  valueConverters.forEach((valueConverter) => {
```
Comment (Pierre-Gilles) [inline server/services/zwavejs-ui/lib/zwaveJSUI.onNodeValueUpdated.js]: Promise.map?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

File: server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js (hunk 1 above)
Comment (sescandell) [reply server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js]: Moved to a function

File: server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js (hunk 2 above)
Comment (sescandell) [reply server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js]: Moved to Promise.map from Bluebird

File: server/services/zwavejs-ui/utils/convertToGladysDevice.js (hunk 3 above)
Comment (sescandell) [reply server/services/zwavejs-ui/utils/convertToGladysDevice.js]: Renamed

File: server/services/zwavejs-ui/lib/zwaveJSUI.onNodeValueUpdated.js (hunk 4 above)
Comment (sescandell) [reply server/services/zwavejs-ui/lib/zwaveJSUI.onNodeValueUpdated.js]: Moved to Promise.map

**Review by Pierre-Gilles — APPROVED** [review]

**Pull request merged**